### DISCUSSION TITLE
I wrote a script which swaps the two foreground workspaces between monitors

### DISCUSSION THREAD

**Discussion by xavierchanth** [nikitabobko/AeroSpace#719] [general]
Edit (March 16, 2025):

Working on a version which uses summon-workspace to be a lot less janky (no extra focuses required).

I'm having issues with getting summon-workspace to work though... it seems it sometimes silently fails... and seems to be monitor dependent.

permalink to this version:

https://github.com/XavierChanth/dotfiles/blob/988c29dbf3d6a9e4ffc602fd568b0e5ebdb15f43/dotfiles/dot-local/bin/aerospace-util.d/swap-workspace-monitors

Original post:

It is a little lengthy, so I've pasted the permalink:

https://github.com/XavierChanth/dotfiles/blob/ab35b4907db15daf55134d4d15ad7402f4f87fec/dotfiles/dot-local/bin/aerospace-util.d/swap-workspace-monitors

The usage is `path/to/script [--wrap-around] (next|prev)`

`up|down|right|left` won't work as it depends on `aerospace move-workspace-to-monitor` which doesn't support that.

It is a little bit slow, as it has to perform a few aerospace operations... what I have now is the bare minimum aerospace commands required.
 
I find myself browsing something on my secondary display and I just quickly want to move it to the main monitor until I find the thing I want to reference, then I swap back.

It also handles the case where a workspace is pinned to a monitor and can't be moved. It will restore the original state in that scenario.

**Comment by mattcoleanderson** [comment]
I love you script. It's functionality I've been looking for. I do however make extensive use of the `workspace-to-monitor-force-assignment` command, which this doesn't work with. I may just have to stop using it.

**Reply by xavierchanth** [reply]
Thanks!

I've designed the script to restore everything to the initial state if a workspace can't be moved. You could end up with some weird states based on which workspace is the pinned one.

Also I believe there are new selectors to move workspaces in the latest version, this was written before those were added, so there may be some enhancements that can be made.

**Comment by awwaiid** [comment]
This may be related to the open PR https://github.com/nikitabobko/AeroSpace/pull/1349 which adds `--when-visible swap` option to `summon-workspace`. I have been using a build with this change for the last 2 months and love it.

**Comment by cwerling** [comment]
Thanks a ton. To get others up to speed more quickly: Only the script behind the **second link** worked for me and I had to use it with `--wrap-around next` (which makes sense, I guess).

**Reply by xavierchanth** [reply]
If someone would like to become the maintainer of the script, I’ve since moved back to Linux. Tired of Apple treating os updates as a marketing event…
